soc/xtensa/intel_adsp/ace/include/intel_ace15_mtpm/adsp_shim.h

```diff
@@ -6,50 +6,67 @@
 #ifndef ZEPHYR_SOC_INTEL_ADSP_SHIM_H_
 #define ZEPHYR_SOC_INTEL_ADSP_SHIM_H_
 
-/* The "shim" block contains most of the general system control
- * registers on cAVS platforms.  While the base address changes, it
- * has remained largely, but not perfectly, compatible between
- * versions.
+#ifndef _ASMLANGUAGE
+
+/**
+ * DfPMCCU
+ * Power Management / Clock Control (HST) Registers
+ *
+ * Offset: 0x0007 1AC0h + 400 0000h * f
+ * These registers block (HST domain) are for general power management
+ * and clock control operation for DSP FW.
  */
+struct ace_dfpmcch {
+	uint32_t dfspsreq;	/* Offset: 0xc0 */
+	uint32_t _unused0[3];
+	uint32_t dfspsrsp;	/* Offset: 0xd0 */
+	uint32_t _unused1[1];
+	uint32_t svcfg;		/* Offset: 0xd8 */
+	uint32_t dfltrc;	/* Offset: 0xdc */
+	uint32_t _unused2[8];
+};
 
-#ifndef _ASMLANGUAGE
-struct cavs_shim {
-	uint32_t skuid;
-	uint32_t _unused0[7];
-	uint32_t dspwc_lo;
-	uint32_t dspwc_hi;
-	uint32_t dspwctcs;
+/**
+ * DfPMCCU
+ * Power Management / Clock Control (ULP) Registers
+ *
+ * Offset: 0x0007 1B00h + 400 0000h * f
```

Comment:
nit, but can we make this comment just:

`Offset: 0x00071B00 + 0x4000000 * f`

what is `f`?

Reply:
f is the all SOC instance number. Actually, f is 0 because f = 0 - HFC -1. Host Function Count actually is 1 => f is 0

Reply:
discussed here: https://github.com/zephyrproject-rtos/zephyr/pull/52343